src/main/kotlin/io/openfuture/snapshot/runner/CommandLineSnapshotRunner.kt

```diff
@@ -3,37 +3,68 @@ package io.openfuture.snapshot.runner
 import com.github.ajalt.clikt.core.CliktCommand
 import com.github.ajalt.clikt.parameters.options.*
 import com.github.ajalt.clikt.parameters.types.int
-import io.openfuture.snapshot.dto.ExportSnapshotRequest
-import io.openfuture.snapshot.exporter.SnapshotExporter
+import io.openfuture.snapshot.dto.SnapshotRequest
+import io.openfuture.snapshot.exporter.CsvSnapshotExporter
+import io.openfuture.snapshot.snapshoter.SnapshotFactory
+import io.openfuture.snapshot.snapshoter.SnapshotMode
+import io.openfuture.snapshot.util.awaitAll
+import java.util.concurrent.Executors
 
-class CommandLineSnapshotRunner(private val snapshotExporter: SnapshotExporter) : CliktCommand(name = "snapshot") {
+class CommandLineSnapshotRunner() : CliktCommand(name = "snapshot") {
 
     private val contractAddress: String? by option(help = "Address of token Smart contract").required()
     private val decimals: Int by option(help = "Snapshot decimals value").int().default(8)
     private val from: Int? by option(help = "Start block number").int().required()
     private val to: Int? by option(help = "End block number").int().required()
     private val nodeAddress: String? by option(help = "Server url of node to connect").required()
     private val filename: String by option(help = "Name of csv file to save").defaultLazy { "snapshot_at_block_$to.csv" }
+    private val snapshotMode: SnapshotMode? by option(help = "Snapshot mode").convert { SnapshotMode.convert(it) }.default(SnapshotMode.ARCHIVED)
+
```

Comment:
I think we need to create common interface for result writers to support various formats (csv, json, xml), and we can easy add support for any format writer

Reply:
I agree with you, but I think there is no need now

Reply:
Andrey is right